gtsam/constrained/BiasedFactor.h

```diff
@@ -0,0 +1,93 @@
+/* ----------------------------------------------------------------------------
+ * GTDynamics Copyright 2020-2021, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * See LICENSE for the license information
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file  BiasedFactor.h
+ * @brief Factor with added bias vector.
+ * @author Yetong Zhang
+ */
+
+#pragma once
+
+#include <gtsam/nonlinear/NonlinearFactor.h>
+
+namespace gtsam {
+
+/** A factor that adds a constant bias term to the original factor.
+ * This factor is used in augmented Lagrangian optimizer to create biased cost
+ * functions.
+ * Note that the noise model is stored twice (both in base factor and the
+ * noisemodel of substitute factor. The noisemodel in the base factor will be
+ * ignored. */
+class GTSAM_EXPORT BiasedFactor : public NoiseModelFactor {
```

Comment:
Based on our discussion, let's strip down to bare minimum and inline it in ALO.cpp